src/shogun/lib/any.h

```diff
@@ -153,7 +153,12 @@ namespace shogun
 		auto compare_impl(maybe_most_important, T* lhs, T* rhs)
 		    -> decltype(lhs->equals(rhs))
 		{
-			return lhs->equals(rhs);
+			if (lhs && rhs)
```

Comment:
@lisitsyn let me know if this is the right place to add those checks

Reply:
Yeah that's a good catch.